app/src/actions/tasks.ts

```diff
@@ -31,15 +33,24 @@ export function loadTasks() {
 export function addTask(task: Task, onComplete: Function) {
   return (dispatch) => {
     delete task._id;
-    return Tasks.add(task)
-      .subscribe((res) => {
-        if (res.length === 1) {
-          dispatch({
-            type: TASK_ADDED,
-            payload: res[0]
-          });
-        }
-      }, null, onComplete); }
+
+    const taskAdd$ = Tasks.add(task).publish();
```

Comment:
I would probably push the publish up into the service - and unless there is a case I am not considering, you can get rid of connect() if, instead of publish, you just use .share() - it's basically a publish/connect (well not connect, refcount, but close enough)